cfgov/ask_cfpb/scripts/migrate_tagging.py

```diff
@@ -96,55 +122,74 @@ def update_feedback_default(stream_value):
     return stream_value
 
 
-def apply_tags(data):
-    """Apply tagging values to answer pages."""
-    category_map = {obj.pk: obj for obj in PortalCategory.objects.all()}
-    migration_user_pk = os.getenv('MIGRATION_USER_PK')
-    user = User.objects.filter(id=migration_user_pk).first()  # default is None
+def update_page(page, entry):
+    initial_status = page.status_string
+    page.user_feedback = update_feedback_default(page.user_feedback)
+    primary_topic = entry.get('primary_topic')
+    topic_ids = entry.get('topics')
+    category_ids = entry.get('categories')
+    if len(topic_ids) > 1:
+        page.primary_portal_topic = TOPIC_OBJECTS.get(primary_topic)
+    for topic_id in topic_ids:
+        page.portal_topic.add(TOPIC_OBJECTS.get(topic_id))
+    for category_id in category_ids:
+        page.portal_category.add(CATEGORY_OBJECTS.get(category_id))
+    page.save_revision(user=USER).publish()
+    if initial_status == 'draft':
+        page.unpublish()
+
+
+def apply_tags(data, singleton=None):
+    """Apply tagging values to answer pages, or to a single Ask ID."""
```

Comment:
Can we update this docstring to something like "Apply tagging values to all answer pages, or to a single set of answer pages determined by their Ask ID"?

Reply:
√